### PR TITLE
EMI Receiver first commit

### LINKED ISSUE #4202: Circuit EMI receiver component needs reporting function
### Description of the feature

The new EMI Receiver component available in Circuit and available in the PyAedt component manager, needs to have a report creation function call.  Something similar to spectralPlot  = circuitObj.post.reports_by_category.spectral(spectralList, solutionName) used for implementing a spectral plot from a transient solution.

### Steps for implementing the feature

Implementation of this feature is dependent on careful consideration of existing PyAedt post implementations.  In addition, the UI provides net selection needed to apply the EMI receiver reporting.  There could be multiple EMI receiver instantiations on the schematic.  The EMI receiver attached nets to be PyAedt detectable, might be important in implementing the report. 

### Useful links and references

image

image

### PR DESCRIPTION
**Pull request by Samuelopez-ansys**
Close #4202

### CODE REVIEW

**Review by SMoraisAnsys — APPROVED** [review]
LGTM

File: pyaedt/modules/PostProcessor.py [hunk 1]
```diff
@@ -726,6 +728,52 @@ def spectral(self, expressions=None, setup_name=None):
             return rep
         return
 
+    @pyaedt_function_handler()
+    def emi_receiver(self, expressions=None, setup_name=None):
+        """Create an EMI receiver Report object.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py:733]:
```suggestion
        """Create an EMI receiver report.
```

File: pyaedt/modules/PostProcessor.py [hunk 2]
```diff
@@ -726,6 +728,52 @@ def spectral(self, expressions=None, setup_name=None):
...
+
+        Parameters
+        ----------
+        expressions : str or list, optional
+            Expression List to add into the report. The expression can be any of the available formula
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py:738]:
```suggestion
            One or more expressions to add into the report. An expression can be any of the formulas that
```
If the type is string or list, I assume that the description should be "One or more", with a developer using a string to add one expression and a list to add more than one?

File: pyaedt/modules/PostProcessor.py [hunk 3]
```diff
@@ -726,6 +728,52 @@ def spectral(self, expressions=None, setup_name=None):
...
+            you can enter into the Electronics Desktop Report Editor.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py:739]:
```suggestion
            can be entered into the Electronics Desktop Report Editor.
```

File: pyaedt/modules/report_templates.py [hunk 4]
```diff
@@ -3786,3 +3786,220 @@ def create(self, plot_name=None):
...
+    @net.setter
+    def net(self, value):
+        if value not in self.available_nets:
+            self.logger.error("Net not available.")
+        else:
+            self._net = value
+
+    @property
+    def band(self):
+        """Net attached to EMI receiver.
```
Comment (PipKat) [inline pyaedt/modules/report_templates.py:3831]:
```suggestion
        """Net attached to the EMI receiver.
```
Should "Net" be "Band"?

**Review by SMoraisAnsys — COMMENTED** [review]
Instead of returning `rep`, would it make more sense to use `res` instead ?

**Comment by Samuelopez-ansys** [comment]
@SMoraisAnsys Probably, it makes more sense, "report", because it is returning a report_templates, what do you think?

File: pyaedt/modules/PostProcessor.py [hunk 5]
```diff
@@ -876,7 +917,8 @@ def available_display_types(self, report_category=None):
             report_category = self.available_report_types[0]
         if report_category:
             return list(self.oreportsetup.GetAvailableDisplayTypes(report_category))
-        return []
+        else:  # pragma: no cover
+            return []
```
Comment (MaxJPRey) [inline pyaedt/modules/PostProcessor.py:921]: You should not need to add the `else` statement.
```suggestion
            return []  # pragma: no cover
```

File: pyaedt/modules/PostProcessor.py [hunk 6]
```diff
@@ -917,8 +917,7 @@ def available_display_types(self, report_category=None):
             report_category = self.available_report_types[0]
         if report_category:
             return list(self.oreportsetup.GetAvailableDisplayTypes(report_category))
-        else:  # pragma: no cover
-            return []
+            return []  # pragma: no cover
```
Comment (MaxJPRey) [inline pyaedt/modules/PostProcessor.py:920]:
```suggestion
        return []  # pragma: no cover
```

Comment (MaxJPRey) [reply pyaedt/modules/PostProcessor.py:920]: My former suggestions was containing a tab issue.

File: pyaedt/modules/PostProcessor.py [hunk 7]
```diff
@@ -935,7 +976,8 @@ def available_quantities_categories(
 
         if solution and report_category and display_type:
             return list(self.oreportsetup.GetAllCategories(report_category, display_type, solution, context))
-        return []
+        else:  # pragma: no cover
+            return []
```
Comment (SMoraisAnsys) [inline pyaedt/modules/PostProcessor.py:980]:
```suggestion
        return []  # pragma: no cover
```

File: pyaedt/modules/PostProcessor.py [hunk 8]
```diff
@@ -1011,7 +1053,8 @@ def available_report_quantities(
                     report_category, display_type, solution, context, quantities_category
                 )
             )
-        return []
+        else:  # pragma: no cover
+            return []
```
Comment (SMoraisAnsys) [inline pyaedt/modules/PostProcessor.py:1057]:
```suggestion
        return []  # pragma: no cover
```

File: pyaedt/modules/PostProcessor.py [hunk 9]
```diff
@@ -1036,7 +1079,8 @@ def available_report_solutions(self, report_category=None):
             report_category = self.available_report_types[0]
         if report_category:
             return list(self.oreportsetup.GetAvailableSolutions(report_category))
-        return None
+        else:  # pragma: no cover
+            return None
```
Comment (SMoraisAnsys) [inline pyaedt/modules/PostProcessor.py:1083]:
```suggestion
        return None  # pragma: no cover
```

File: pyaedt/modules/PostProcessor.py [hunk 10]
```diff
@@ -2206,7 +2250,8 @@ def _get_base_name(self, setup):
                     if sol["ID"] == setups_data[setup]["SolutionId"]:
                         base_name += " : " + sol["Name"]
                         return base_name
-        return ""
+        else:  # pragma: no cover
+            return ""
```
Comment (SMoraisAnsys) [inline pyaedt/modules/PostProcessor.py:2254]:
```suggestion
        return ""  # pragma: no cover
```

File: pyaedt/modules/PostProcessor.py [hunk 11]
```diff
@@ -2217,7 +2262,8 @@ def _get_intrinsic(self, setup):
             for intr in intrinsics:
                 if isinstance(intr, list) and len(intr) == 2:
                     intr_dict[intr[0]] = intr[1].replace("\\", "").replace("'", "")
-        return intr_dict
+        else:  # pragma: no cover
+            return intr_dict
```
Comment (SMoraisAnsys) [inline pyaedt/modules/PostProcessor.py:2266]:
```suggestion
        return intr_dict  # pragma: no cover
```

File: pyaedt/modules/report_templates.py [hunk 12]
```diff
@@ -3786,3 +3786,221 @@ def create(self, plot_name=None):
...
+        return self._emission
+
+    @emission.setter
+    def emission(self, value):
+        if value == "CE":
+            self._emission = value
+            self.props["context"]["emission"] = "0"
+        else:
+            self._emission = "RE"
+            self.props["context"]["emission"] = "1"
```
Comment (SMoraisAnsys) [inline pyaedt/modules/report_templates.py:3862]:
```suggestion
        if value == "CE":
            self._emission = value
            self.props["context"]["emission"] = "0"
        elif value == "RE":
            self._emission = value
            self.props["context"]["emission"] = "1"
        else:
            self.logger.error("Emission must be 'CE' or 'RE', value '{}' is not valid.".format(value))
```

**Review by SMoraisAnsys — DISMISSED** [review]
Minor changes to add log error and to follow @MaxJPRey previous comments.

**Review by SMoraisAnsys — APPROVED** [review]
LGTM, thanks for taking into account every review !

**Pull request merged**